### PR TITLE
Codex/add xenium annotation merge

### PR DESCRIPTION
**Pull request by mthimma**
When we convert anndata, we avoid overwriting existing `"cell_id"`. https://github.com/Taylor-CCB-Group/MDV/issues/438.

We have also added a new cli command for simultaneously 
- patching this column in spatial projects when necessary, based on re-reading it from the original table data.
- adding additional columns of annotation joined on the `cell_id` in an annotation CSV file.

## Summary by CodeRabbit

* **New Features**
  * New CLI command to patch spatial annotations from CSV with selectable datasource/table/instance-key mapping, delimiter and missing-value options, dry-run vs apply, and optional overwrite of preserved keys; creates backups when applying.
* **Bug Fixes**
  * Improved cell identifier generation during conversion to avoid identifier name collisions.
* **Tests**
  * Added comprehensive tests covering patching behavior, idempotency, validation, and conversion edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Taylor-CCB-Group/MDV/pull/444)

### CODE REVIEW

File: python/mdvtools/spatial/annotations.py [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+        ]
+        project.set_column(
+            datasource,
+            {
+                "name": column,
+                "field": column,
+                "datatype": "text",
+            },
+            values,
+        )
```
[the remark this answers is not in the document]
Comment (xinaesthete) [reply python/mdvtools/spatial/annotations.py:482]: We probably don't always want these to always be `"text"`, the user might be trying to add something like UMAP annotations.

[the remark this answers is not in the document]
Comment (xinaesthete) [reply python/mdvtools/spatial/annotations.py:482]: Yes really we should look more carefully at `MDVProject.add_annotations`, I think codex was being sniffy about it for some reason, but if there are problems with how it's implemented we should consider fixing them rather than adding another method for doing almost exactly the same thing.

**Comment by xinaesthete** [comment]
I'm not sure why the TypeScript CI is failing here. I'd accidentally pushed a bad commit straight to main, which may be what this is based off.

File: python/mdvtools/spatial/annotations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xinaesthete) [reply python/mdvtools/spatial/annotations.py:482]: @coderabbitai can you create an issue for this?
